pypesto/sample/parallel_tempering.py

```diff
@@ -164,6 +173,72 @@ def swap_samples(self) -> Sequence[bool]:
     def adjust_betas(self, i_sample: int, swapped: Sequence[bool]):
         """Adjust temperature values. Default: Do nothing."""
 
+    def compute_log_evidence(
+        self, result: Result, method: str = "trapezoid"
+    ) -> float:
+        """Perform thermodynamic integration to estimate the log evidence.
+
+        Parameters
+        ----------
+        result:
+            Result object containing the samples.
+        method:
+            Integration method, either 'trapezoid' or 'simpson' (uses scipy for integration).
+        """
+        from scipy.integrate import simpson, trapezoid
+
+        if self.options["beta_init"] == "exponential_decay":
+            logger.warning(
+                "The temperature schedule is not optimal for thermodynamic integration. "
+                "Carefully check the results. Consider using beta_init='beta_decay' for better results."
+            )
+
+        burn_in = result.sample_result.burn_in
+        trace_loglike = (
+            result.sample_result.trace_neglogprior[::-1, burn_in:]
+            - result.sample_result.trace_neglogpost[::-1, burn_in:]
+        )
+        mean_loglike_per_beta = np.mean(trace_loglike, axis=1)
+        if method == "trapezoid":
+            log_evidence = trapezoid(
+                y=mean_loglike_per_beta, x=self.betas[::-1]
+            )
+        elif method == "simpson":
+            log_evidence = simpson(
+                y=mean_loglike_per_beta, x=self.betas[::-1], even="last"
+            )
+        else:
+            raise ValueError(
+                f"Unknown method {method}. Choose 'trapezoid' or 'simpson'."
+            )
+
+        return log_evidence
+
+
+def beta_decay_for_beta(n_chains: int, alpha: float) -> np.ndarray:
```

Comment:
Maybe change function name to `beta_decay_betas` to be more similar and consistent with the other function name `near_exponential_decay_betas`.